### PR TITLE
Fix handling of top level nulls in the input to Reduce function

### LINKED ISSUE #5768: Error for NULL array handling in Reduce with coalesce as the output function
### Bug description

`NULL` array and array with null elements are not the same. For `NULL` array, it may should not be passed to input/output function to avoid incorrect update of final result, like `coalesce` function.
This issue can be reproduced by UT like:
```
TEST_F(ReduceTest, nullArray) {
  auto arrayVector = makeNullableArrayVector<int64_t>(
      {std::nullopt, {{1, std::nullopt}}, {{std::nullopt, 2}}, {{1, 2, 3}}});
  auto data = makeRowVector({arrayVector});
  auto result = evaluate(
      "reduce(c0, 0, (s, x) -> s + x, s -> coalesce(s, 0) * 10)",
      data);
  assertEqualVectors(
      makeNullableFlatVector<int64_t>({std::nullopt, 0, 0, 60}), result);
}
```

### System information

N/A

### Relevant logs

_No response_

### PR DESCRIPTION
**Pull request by yma11**
Fixes #5768 
Fix handling of top level nulls in the input to Reduce function. Currently, the nulls are being passed as rows to be operated on to the lambda functions in Reduce. This patch fixes this by ensuring only the non null Rows are passed to the lambda and nulls are added to the output.

Test Plan:
Adds a unit test for Reduce function.

### CODE REVIEW

**Comment by FelixYBW** [comment]
@rui-mo

File: velox/functions/prestosql/Reduce.cpp [hunk 1]
```diff
@@ -81,13 +81,14 @@ class ReduceFunction : public exec::VectorFunction {
     const auto& initialState = args[1];
     auto partialResult =
         BaseVector::create(initialState->type(), rows.end(), context.pool());
-
+    std::vector<bool> isNullArrays(rows.end(), false);
```
Comment (winningsix) [inline velox/functions/prestosql/Reduce.cpp:84]: Minor: ```isNulls``` better?

File: velox/functions/prestosql/Reduce.cpp [hunk 2]
```diff
@@ -169,6 +170,11 @@ class ReduceFunction : public exec::VectorFunction {
           &localResult);
     }
     context.moveOrCopyResult(localResult, rows, result);
+    for (int i = 0; i < rows.end(); i++) {
+      if (isNullArrays[i]) {
+        result->setNull(i, true);
```
Comment (winningsix) [inline velox/functions/prestosql/Reduce.cpp:175]: ```result->setNull(i, isNullArrays[i])```

File: velox/functions/prestosql/Reduce.cpp [hunk 3]
```diff
@@ -81,13 +81,14 @@ class ReduceFunction : public exec::VectorFunction {
...
         BaseVector::create(initialState->type(), rows.end(), context.pool());
-
+    std::vector<bool> isNullArrays(rows.end(), false);
     // Process null and empty arrays.
     auto* rawNulls = flatArray->rawNulls();
     auto* rawSizes = flatArray->rawSizes();
     rows.applyToSelected([&](auto row) {
       if (rawNulls && bits::isBitNull(rawNulls, row)) {
         partialResult->setNull(row, true);
+        isNullArrays[row] = true;
```
Comment (rui-mo) [inline velox/functions/prestosql/Reduce.cpp:91]: Partial result is set as null for this row, do we still need `isNullArrays` to record the nullability?

File: velox/functions/prestosql/Reduce.cpp [hunk 4]
```diff
@@ -169,6 +170,11 @@ class ReduceFunction : public exec::VectorFunction {
           &localResult);
     }
     context.moveOrCopyResult(localResult, rows, result);
+    for (int i = 0; i < rows.end(); i++) {
```
Comment (rui-mo) [inline velox/functions/prestosql/Reduce.cpp:173]: Does something like below work?
```
for (const auto& row : rows) {
  if (partialResult->isNull(row) {
      result->setNull(i, true);
   }
}
```

File: velox/functions/prestosql/Reduce.cpp (hunk 3 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:91]: Partial result is set as null but it can't distinguish Null Array and Array with null element in later functions, thus cause the the result is same for these 2 cases and not correct. So we make a special marker for Null Array and apply this maker on the final result.

File: velox/functions/prestosql/Reduce.cpp (hunk 4 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:173]: I tried this. It doesn't work and will cause wrong result for the not NULL arrays.

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:173]: It doesn't work, seems partialResult will changed during merge function and complete function compute.

File: velox/functions/prestosql/Reduce.cpp (hunk 2 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:175]: Maybe `setNull(i, true)` is more clear for code reading.

File: velox/functions/prestosql/Reduce.cpp (hunk 3 above)
Comment (rui-mo) [reply velox/functions/prestosql/Reduce.cpp]: > it can't distinguish Null Array and Array with null element.

Is it the expected behavior by setting null for both null array and array with null element?

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: In this line, `partialResult` is only setting null true for NULL Array. For array with null element, the `setNull` should be done during merge function process. The problem is that even when `partialResult` is setting null true there, this value will be erased during finish function like `coalesce(s, 0)`, which will change `null` value to `0`, but it shouldn't apply on NULL array.

Comment (rui-mo) [reply velox/functions/prestosql/Reduce.cpp]: Is above the expected behavior of Spark? I noticed this function is a presto one, and are they the same?

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: This is expected behavior of Spark. But no matter for Spark or Presto, NULL array handle should be the same.

**Comment by yma11** [comment]
@mbasmanova can you help take a look at this fix? Thanks.

**Review by mbasmanova — COMMENTED** [review]
@bikramSingh91 Bikram, would you help review this PR?

**Comment by bikramSingh91** [comment]
The root cause for this issue is that we are also evaluating null rows by passing them to the lambda, however, we should do something similar to how we handle nulls for null propagating functions in Expr.cpp, that is, we only evaluate null rows in the lambdas and add nulls to the final result at the end. We can do this by creating a LocalSelectivityVector applyRows and set it correctly with non Null rows (see this for reference: https://github.com/facebookincubator/velox/blob/main/velox/expression/FieldReference.cpp#L56-L63 )
Then use this moving forward from L103 and L160. Finally we add the nulls to 'result' using result->addNulls(flatArray->rawNulls(), rows);

A question might arise here as to why we don’t just leverage the default null behavior handling in expr.cpp, a good reason for that is explained here: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/Reduce.cpp#L57-L63

Another question can be is why we don’t just add nulls at the end, the reason for that is that we don’t even want those rows to be evaluated in case the lambdas run into issues on those rows when they are not even supposed to evaluate them, also it saves extra cycles for us by not evaluating those rows.

**Comment by yma11** [comment]
> The root cause for this issue is that we are also evaluating null rows by passing them to the lambda, however, we should do something similar to how we handle nulls for null propagating functions in Expr.cpp, that is, we only evaluate null rows in the lambdas and add nulls to the final result at the end. We can do this by creating a LocalSelectivityVector applyRows and set it correctly with non Null rows (see this for reference: https://github.com/facebookincubator/velox/blob/main/velox/expression/FieldReference.cpp#L56-L63 ) Then use this moving forward from L103 and L160. Finally we add the nulls to 'result' using result->addNulls(flatArray->rawNulls(), rows);
> 
> A question might arise here as to why we don’t just leverage the default null behavior handling in expr.cpp, a good reason for that is explained here: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/Reduce.cpp#L57-L63
> 
> Another question can be is why we don’t just add nulls at the end, the reason for that is that we don’t even want those rows to be evaluated in case the lambdas run into issues on those rows when they are not even supposed to evaluate them, also it saves extra cycles for us by not evaluating those rows.

Thanks for comments and hints. I updated so please help review again.

File: velox/functions/prestosql/Reduce.cpp [hunk 5]
```diff
@@ -192,4 +207,4 @@ VELOX_DECLARE_VECTOR_FUNCTION(
     ReduceFunction::signatures(),
     std::make_unique<ReduceFunction>());
 
-} // namespace facebook::velox::functions
+} // namespace facebook::velox::functions
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:210]: nit: can you add the removed newline back?

File: velox/functions/prestosql/tests/ReduceTest.cpp [hunk 6]
```diff
@@ -125,7 +125,7 @@ TEST_F(ReduceTest, finalSelection) {
       "reduce(c0, 10, (s, x) -> s + x, s -> row_constructor(s)))",
       input);
 
-  auto expectedResult = makeRowVector({makeFlatVector<int64_t>(
+  auto expectedResult = makeFlatVector<int64_t>(
```
Comment (bikramSingh91) [inline velox/functions/prestosql/tests/ReduceTest.cpp:128]: nit: can you revert these unrelated changes

File: velox/functions/prestosql/tests/ReduceTest.cpp [hunk 7]
```diff
@@ -205,3 +205,13 @@ TEST_F(ReduceTest, finalSelectionLargerThanInput) {
 
   assertEqualVectors(makeFlatVector<int64_t>({2, 1}), result);
 }
+
+TEST_F(ReduceTest, nullArray) {
+  auto arrayVector = makeNullableArrayVector<int64_t>(
```
Comment (bikramSingh91) [inline velox/functions/prestosql/tests/ReduceTest.cpp:210]: nit: can you please add a comment on what this test is verifying?

File: velox/functions/prestosql/Reduce.cpp [hunk 8]
```diff
@@ -67,39 +67,47 @@ class ReduceFunction : public exec::VectorFunction {
...
+    exec::LocalSelectivityVector nonNullRowsHolder(*context.execCtx());
+    const SelectivityVector* nonNullRows = &rows;
+    DecodedVector decodedArray;
+    decodedArray.decode(*args[0], rows);
+    if (decodedArray.mayHaveNulls()) {
+      nonNullRowsHolder.get(rows);
+      nonNullRowsHolder->deselectNulls(
+          decodedArray.nulls(), rows.begin(), rows.end());
+      nonNullRows = nonNullRowsHolder.get();
+      auto selected = nonNullRows->countSelected();
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:84]: can you please remove these unused variables?

File: velox/functions/prestosql/Reduce.cpp [hunk 9]
```diff
@@ -168,7 +178,12 @@ class ReduceFunction : public exec::VectorFunction {
           nullptr,
           &localResult);
     }
-    context.moveOrCopyResult(localResult, rows, result);
+    context.moveOrCopyResult(localResult, *nonNullRows, result);
+    for (int i = 0; i < flatArray->size(); i++) {
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:182]: this loop can be replaced with 
`result->addNulls(flatArray->rawNulls(), rows);`

File: velox/functions/prestosql/Reduce.cpp [hunk 10]
```diff
@@ -155,8 +163,10 @@ class ReduceFunction : public exec::VectorFunction {
     }
 
     // Apply output function.
-    VectorPtr localResult;
-    auto outputFuncIt = args[3]->asUnchecked<FunctionVector>()->iterator(&rows);
+    VectorPtr localResult =
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:166]: no need to create a new basevector here, expression eval will create an appropriate one if required

File: velox/functions/prestosql/Reduce.cpp [hunk 11]
```diff
@@ -67,39 +67,47 @@ class ReduceFunction : public exec::VectorFunction {
...
+    nonNullRows->applyToSelected([&](auto row) {
+      if (rawSizes[row] == 0) {
         partialResult->copy(initialState.get(), row, row, 1);
       }
     });
 
     // Make sure already populated entries in 'partialResult' do not get
     // overwritten if 'arrayRows' shrinks in subsequent iterations.
-    const SelectivityVector& validRowsInReusedResult = rows;
+    // const SelectivityVector& validRowsInReusedResult = rows;
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:104]: nit: can you remove uncommented code and move L106 here instead?

File: velox/functions/prestosql/Reduce.cpp [hunk 12]
```diff
@@ -67,39 +67,47 @@ class ReduceFunction : public exec::VectorFunction {
...
       const SelectivityVector& rows,
       std::vector<VectorPtr>& args,
-      const TypePtr& /* outputType */,
+      const TypePtr& outputType,
       exec::EvalCtx& context,
       VectorPtr& result) const override {
     VELOX_CHECK_EQ(args.size(), 4);
-
+    // Identify the rows need to be computed.
+    exec::LocalSelectivityVector nonNullRowsHolder(*context.execCtx());
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:75]: You can calculate the nonNullRows after creating the flatArray on line 90 using the flatArray instead of the decoded vector. Also, please maintain the use of exec::LocalDecodedVector in the original code instead of using the DecodedVector class here directly.

File: velox/functions/prestosql/Reduce.cpp (hunk 9 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:175]: It doesn't work. Maybe because we passed nonNullRows to lambda and can't apply flatArray `rawNulls` here.

File: velox/functions/prestosql/Reduce.cpp (hunk 12 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:79]: updated.

File: velox/functions/prestosql/tests/ReduceTest.cpp (hunk 6 above)
Comment (yma11) [reply velox/functions/prestosql/tests/ReduceTest.cpp:128]: Sorry seems I pushed a dirty commit by mistake.

Comment (yma11) [reply velox/functions/prestosql/tests/ReduceTest.cpp:128]: Not actually. After changes in `Reduce` function, this `RowType` output is different from the result. The inner layer is same but for `RowType`, the expected one is like `{null}, {1}, {2}....` while actual result is `null, {1}, {2}....`

File: velox/functions/prestosql/Reduce.cpp (hunk 8 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: Sorry seems I pushed a dirty commit by mistake.

**Comment by yma11** [comment]
@bikramSingh91 can you help take a further look? Thanks.

File: velox/functions/prestosql/Reduce.cpp (hunk 9 above)
Comment (bikramSingh91) [reply velox/functions/prestosql/Reduce.cpp:175]: @yma11 in that case, you can do 

```
partialResult->addNulls(flatArray->rawNulls(), rows);
context.moveOrCopyResult(localResult, rows, result);
```

since partialResult has already been initialized with rows.end() size.

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:175]: Tried but some TCs failed. Besides it doesn't always make sense to apply `addNulls` on `partialResult`. Take `coalesce(s, 0)` for example, the null value will be incorrectly updated to `0` in this case.

Comment (bikramSingh91) [reply velox/functions/prestosql/Reduce.cpp:175]: @yma11 
> Tried but some TCs failed. 

Curious which test cases failed and why? I might be missing something here so would love to know.

> Besides it doesn't always make sense to apply addNulls on partialResult. Take coalesce(s, 0) for example, the null value will be eventually incorrectly updated to 0 in this case.

The nulls we are adding here is in the last step and are the nulls rows in the array which should not be passed to the lambda. these should then we copied over directly to the final result in moveOrCopyResult() and should end up as null. Curious whats causing the result to be overwritten in the coalesce example you mentioned.

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:175]: > Curious which test cases failed and why? I might be missing something here so would love to know.
> 
The existing TCs `elementIndicesOverwrite` and `try` fail with errors like: 
```
Value of: expected->equalValueAt(actual.get(), i, i)
  Actual: false
Expected: true
at 0: expected -2, but got null
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0906 15:44:39.674255 419537 Exceptions.h:69] Line: ../.././velox/vector/ConstantVector.h:276, Function:addNulls, Expression:  addNulls not supported, Source: RUNTIME, ErrorCode: INVALID_STATE
unknown file: Failure
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: addNulls not supported
Retriable: False
```
> The nulls we are adding here is in the last step and are the nulls rows in the array which should not be passed to the lambda. these should then we copied over directly to the final result in moveOrCopyResult() and should end up as null. Curious whats causing the result to be overwritten in the coalesce example you mentioned.
```
partialResult->addNulls(flatArray->rawNulls(), rows);
context.moveOrCopyResult(localResult, rows, result);
```
Above lines will guarantee `partialResult` will have correct NULL values. But I think it doesn't distinguish it's a null value or actually a Null array. Take UT in this PR as example, `reduce(c0, 0, (s, x) -> s + x, s -> coalesce(s, 0) * 10)`, partialResult will be null for both an array like `(1, 2, null, value)` and NULL array. And in the finish function ` coalesce(s, 0)`, the null in `partialResult` will become 0 not matter what it represents.  This is what this PR wants to address.

File: velox/functions/prestosql/tests/ReduceTest.cpp (hunk 6 above)
Comment (bikramSingh91) [reply velox/functions/prestosql/tests/ReduceTest.cpp:128]: @yma11 `null, {1}, {2}.... ` is what should be expected. evaluate<> returns the result of the expression which should have the null at the top level rowVector layer and this change to check the child of the row is actually hiding this bug. The expression used in the test  `if (c1 < 100, row_constructor(c1), reduce(c0, 10, (s, x) -> s + x, s -> row_constructor(s)))` adds a top level null to the result via the first branch  `row_constructor(c1)` which means the mismtach` {null} != null ` is a valid bug. This is happening because in this change the following piece of code is overwriting the row in Reduce which should not have been touched. 
```
   for (int i = 0; i < flatArray->size(); i++) {
      if (flatArray->isNullAt(i)) {
        result->setNull(i, true);
      }
    }
```
Please see explanation of why this is happening in my other comment.
Therefore, to fix this test we need to add top level nulls to  expectedResult at the RowVector layer.

File: velox/functions/prestosql/Reduce.cpp (hunk 9 above)
Comment (bikramSingh91) [reply velox/functions/prestosql/Reduce.cpp:175]: @yma11 I looked into this issue and noticed an unexpected pre-existing API implementation + a bug in this PR

(also to clarify, I mistakingly mentioned using `partialResult` when adding nulls, instead of using `localResult`)

**1. unexpected pre-existing API implementation:**

I was curious why BaseVector::addNulls is not working properly to I dug into its implementation. Turns out if `flatArray->rawNulls()` is nullptr then it treats the unselected rows in `rows` as nulls and adds those nulls to the vector. This is not clear from the method comment. I'll look into where else this might be causing bugs later. The expectation here is that when using this API, we should add all the nulls in `flatArray->rawNulls()` for rows that are active in `rows`. So, to ensure we do that, please use this instead:

```
   if (flatArray->rawNulls()) {
      localResult->addNulls(flatArray->rawNulls(), rows);
    }
    context.moveOrCopyResult(localResult, rows, result);
```

**2. a bug in this PR**

if we proceed with the current implementation of using this:
```
   for (int i = 0; i < flatArray->size(); i++) {
      if (flatArray->isNullAt(i)) {
        result->setNull(i, true);
      }
    }
```
This would touch rows that are not marked as "active" in `rows` which will cause incorrect results as those rows might be filled in by other branches of the expression tree. This bug was exposed by investigating failure in `TEST_F(ReduceTest, finalSelection)` (see comment there on how to fix the test after applying the suggested implementation as suggested above.

File: velox/functions/prestosql/tests/ReduceTest.cpp (hunk 6 above)
Comment (yma11) [reply velox/functions/prestosql/tests/ReduceTest.cpp]: Done.

File: velox/functions/prestosql/Reduce.cpp (hunk 9 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: Thanks for finding this out. The `addNulls` works as you suggested. Please take a look again.

**Review by bikramSingh91 — APPROVED** [review]

**Comment by bikramSingh91** [comment]
@yma11 Can you please check the circleCI failures to ensure any legit failures are addressed before merging.
@kgpai @kagamiori Can either of you ensure this gets merged.

**Comment by yma11** [comment]
> @yma11 Can you please check the circleCI failures to ensure any legit failures are addressed before merging. @kgpai @kagamiori Can either of you ensure this gets merged.
@bikramSingh91 updated. Thanks for your review!

**Comment by yma11** [comment]
@kgpai @kagamiori Can you guys help merge this PR?

File: velox/functions/prestosql/Reduce.cpp [hunk 13]
```diff
@@ -168,7 +172,10 @@ class ReduceFunction : public exec::VectorFunction {
           nullptr,
           &localResult);
     }
-    context.moveOrCopyResult(localResult, rows, result);
+    if (flatArray->rawNulls()) {
+      localResult->addNulls(flatArray->rawNulls(), rows);
+    }
+    context.moveOrCopyResult(localResult, *nonNullRows, result);
```
Comment (bikramSingh91) [inline velox/functions/prestosql/Reduce.cpp:178]: this should use `rows` instead of nonNullRows

File: velox/functions/prestosql/tests/ReduceTest.cpp [hunk 14]
```diff
@@ -205,3 +205,15 @@ TEST_F(ReduceTest, finalSelectionLargerThanInput) {
 
   assertEqualVectors(makeFlatVector<int64_t>({2, 1}), result);
 }
+
```
Comment (bikramSingh91) [inline velox/functions/prestosql/tests/ReduceTest.cpp:208]: curious, how is the test case `finalSelection` passing now?

**Review by bikramSingh91 — CHANGES_REQUESTED** [review]

File: velox/functions/prestosql/Reduce.cpp (hunk 13 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: Done.

File: velox/functions/prestosql/tests/ReduceTest.cpp (hunk 14 above)
Comment (yma11) [reply velox/functions/prestosql/tests/ReduceTest.cpp:208]: Recently there are some PRs for Null handling in functions are merged like 6391. Maybe they introduced changes affecting this case.

File: velox/functions/prestosql/Reduce.cpp (hunk 13 above)
Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp]: Seems if we use `*nonNullRows` here, we don't need change UT `finalSelection`.

**Review by bikramSingh91 — APPROVED** [review]

**Comment by yma11** [comment]
@bikramSingh91 Seems there is a Linter issue. Need to look at it?

File: velox/functions/prestosql/Reduce.cpp [hunk 15]
```diff
@@ -168,6 +172,9 @@ class ReduceFunction : public exec::VectorFunction {
           nullptr,
           &localResult);
     }
+    if (flatArray->rawNulls()) {
+      localResult->addNulls(flatArray->rawNulls(), rows);
```
Comment (mbasmanova) [inline velox/functions/prestosql/Reduce.cpp:176]: @yma11 I'm seeing errors that seem to be coming from this change:

```
VeloxRuntimeError:  addNulls not supported presto.default.reduce
	at Unknown.# 0  facebook::velox::VeloxException::VeloxException(char const*, unsigned long, char const*, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, std::basic_string_view<char, std::char_traits<char> >, bool, facebook::velox::VeloxException::Type, std::basic_string_view<char, std::char_traits<char> >)(Unknown Source)
	at Unknown.# 1  void facebook::velox::detail::veloxCheckFail<facebook::velox::VeloxRuntimeError, char const*>(facebook::velox::detail::VeloxCheckFailArgs const&, char const*)(Unknown Source)
	at Unknown.# 2  facebook::velox::ConstantVector<facebook::velox::ComplexType>::addNulls(unsigned long const*, facebook::velox::SelectivityVector const&)(Unknown Source)
	at Unknown.# 3  facebook::velox::functions::(anonymous namespace)::ReduceFunction::apply(facebook::velox::SelectivityVector const&, std::vector<std::shared_ptr<facebook::velox::BaseVector>, std::allocator<std::shared_ptr<facebook::velox::BaseVector> > >&, std::shared_ptr<facebook::velox::Type const> const&, facebook::velox::exec::EvalCtx&, std::shared_ptr<facebook::velox::BaseVector>&) const(Unknown Source)
```

Comment (yma11) [reply velox/functions/prestosql/Reduce.cpp:176]: Found that this is already resolved by [6886](https://github.com/facebookincubator/velox/pull/6886). Thanks @bikramSingh91 for the fix.
